### ISSUE TITLE
export-api-def fails with path not a string error

### ISSUE THREAD

**Issue by jannyHou** [strongloop/loopback-cli#27]
Moved from https://github.com/strongloop/generator-loopback/issues/276
cc @qpresley 
# Description/Steps to reproduce
Steps to reproduce are very simple.  Create a loopback app using the `lb` CLI, chose the empty server app.  When it completes run `lb export-api-def -o ./testApp.yaml` and you will get the error.  Here is the output of my local run

```
The API Connect team at IBM happily continues to develop,
support and maintain LoopBack, which is at the core of
API Connect. When your APIs need robust management and
security options, please check out http://ibm.biz/tryAPIC

Quentins-MBP:testApp qpresley$ lb export-api-def -o ./testApp.yaml
FILEPATH:  true
events.js:141
      throw er; // Unhandled 'error' event
      ^

TypeError: path must be a string
    at TypeError (native)
    at Object.fs.openSync (fs.js:549:18)
    at Object.fs.writeFileSync (fs.js:1156:15)
    at module.exports.yeoman.Base.extend.generate (/Users/qpresley/.nvm/versions/node/v4.7.2/lib/node_modules/loopback-cli/node_modules/generator-loopback/export-api-def/index.js:45:10)
    at Object.<anonymous> (/Users/qpresley/.nvm/versions/node/v4.7.2/lib/node_modules/loopback-cli/node_modules/generator-loopback/node_modules/yeoman-generator/lib/base.js:431:23)
    at /Users/qpresley/.nvm/versions/node/v4.7.2/lib/node_modules/loopback-cli/node_modules/generator-loopback/node_modules/yeoman-generator/node_modules/run-async/index.js:25:25
    at /Users/qpresley/.nvm/versions/node/v4.7.2/lib/node_modules/loopback-cli/node_modules/generator-loopback/node_modules/yeoman-generator/node_modules/run-async/index.js:24:19
    at /Users/qpresley/.nvm/versions/node/v4.7.2/lib/node_modules/loopback-cli/node_modules/generator-loopback/node_modules/yeoman-generator/lib/base.js:432:9
    at processImmediate [as _immediateCallback] (timers.js:396:17)
```

**Comment by crandmck** [comment]
Definitely broken in latest version (2.2.0).  

The error does not occur without the option (i.e. just `lb export-api-def`) and the YAML is sent to stdout. Since you can redirect it to a file, that is a  workaround at least.

**Comment by qpresley** [comment]
According to Janny's tests it also doesn't fail using `slc loopback:export-api-def -o someApp.yaml`

**Comment by bajtos** [comment]
I suspect this problem may be caused by the glue code in loopback-cli that's setting up Yeoman's environment. Since I am the author of that code, I may be the best person to work on this issue. cc @ritch

**Comment by enricop89** [comment]
@bajtos any updates?

**Comment by cfjedimaster** [comment]
As an aside, this is kind of a big deal if you want to use the new native API management stuff in BM. (That's how I ran into it.)

**Comment by bajtos** [comment]
I managed to track down the problem and find a reasonably easy fix - see #34. @enricop89 @cfjedimaster can you give it a try? 

I think the feature branch can be installed by running the following command:

```
npm install -g strongloop/loopback-cli#fix/export-api-def
```

**Comment by enricop89** [comment]
@bajtos I installed the fix/export-api-def branch and now it works (with the option -o)! 🏅 
Thank you!

**Comment by bajtos** [comment]
Thanks!

I have released `loopback-cli@2.5.1` with the fix.

**Comment by cfjedimaster** [comment]
I can confirm it works. Thank you.
